### PR TITLE
auth: improve metadata caching

### PR DESCRIPTION
**Pull request by mind04**
### Short description
The number of yaks in the metadata cache was impressive. This pull will hopefully make them an extinct species :)

### Checklist
I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by Habbie** [comment]
After almost a full day of running tests in a loop, this PR has proven stable for the caching issues we have on master.

**Comment by RobinGeuze** [comment]
Might be a good idea to add the following test case (since this was a non-racy variant of the bug):
```
cleannsupdate <<!
server $nameserver $port
zone test.dyndns
update DELETE test.dyndns. NSEC3PARAM
update add a.host.test.dyndns. 3600 CNAME localhost.
send
answer
!
```
It will process the NSEC3PARAM as usual and then error out in the update add, because the a.host.test.dyndns. label already has an A record. This will then cause a transaction rollback. With the old code the NSEC3PARAM being deleted ended up in the cache regardless, which caused the obvious rectify issues.

**Review by Habbie — COMMENTED** [review]
I found this comment stuck in my review queue. Posting it now, will do a more complete review soon.

File: pdns/dbdnsseckeeper.cc [hunk 1]
```diff
@@ -47,6 +47,7 @@ using namespace boost::assign;
 
 DNSSECKeeper::keycache_t DNSSECKeeper::s_keycache;
 DNSSECKeeper::metacache_t DNSSECKeeper::s_metacache;
+int64_t DNSSECKeeper::s_metaCacheCleanActions = 0;
```
[the remark this answers is not in the document]
Comment (mind04) [reply pdns/dbdnsseckeeper.cc:50]: Not necessary, the counter is never updated / accessed without the appropriate s_metacachelock

[the remark this answers is not in the document]
Comment (Habbie) [reply pdns/dbdnsseckeeper.cc:50]: ah! yes

**Pull request merged**